pypdf/_writer.py

```diff
@@ -1796,12 +1806,8 @@ def addNamedDestination(
 
     def remove_links(self) -> None:
         """Remove links and annotations from this output."""
-        pg_dict = cast(DictionaryObject, self.get_object(self._pages))
-        pages = cast(ArrayObject, pg_dict[PA.KIDS])
-        for page in pages:
-            page_ref = cast(DictionaryObject, self.get_object(page))
-            if PG.ANNOTS in page_ref:
-                del page_ref[PG.ANNOTS]
+        for page in self.pages:
+            self.remove_objects_from_page(page, ObjectDeletionFlag.ALL_ANNOTS)
```

Comment:
I'm uncertain here, but I think it should either be `ObjectDeletionFlag.LINKS` or `ObjectDeletionFlag.ALL_ANNOTS | ObjectDeletionFlag.LINKS`